protostar/starknet_gateway/gateway_facade.py

```diff
@@ -168,15 +172,18 @@ async def declare(
             },
         )
 
-        result = await self._gateway_client.declare(tx, token)
-        register_response(dataclasses.asdict(result))
-        if wait_for_acceptance:
-            if self._logger:
-                self._logger.info("Waiting for acceptance...")
-            _, status = await self._gateway_client.wait_for_tx(
-                result.transaction_hash, wait_for_accept=wait_for_acceptance
-            )
-            result.code = status
+        try:
+            result = await self._gateway_client.declare(tx, token)
+            register_response(dataclasses.asdict(result))
+            if wait_for_acceptance:
+                if self._logger:
+                    self._logger.info("Waiting for acceptance...")
+                _, status = await self._gateway_client.wait_for_tx(
+                    result.transaction_hash, wait_for_accept=wait_for_acceptance
+                )
+                result.code = status
+        except TransactionFailedError as ex:
+            raise TransactionException(str(ex)) from ex
```

Comment:
Shouldn't that be ex.message?

Reply:
`str(ex)` decorates `ex.message` with information about exception kind